cc/Strategy2/Attacker.cpp

```diff
@@ -11,11 +11,53 @@ void Attacker::execute_behavior(Attacker::Behavior behavior) {
 		case Behavior::UVF_To_Goal:
 			uvf_to_goal({0,0});
 			break;
+
 	}
 }
 
 void Attacker::uvf_to_goal(Point ball) {
 	Vector ball_to_goal = goal_center - ball;
-	go_to_pose(ball, ball_to_goal);
-//	Inserir corrida para o gol ao chegar na bola
+
+	if(distance(ball,get_pose().position)>0.03){
+		go_to_pose(ball, ball_to_goal);
+	}else{
+	go_in_direction(ball_to_goal);
+	}
+	//	Inserir corrida para o gol ao chegar na bola
+}
+
+void Attacker::spin_shot(bool clockwise){
+	if (clockwise){
+		spin(-1.4);// Robô gira no sentido horário
+	}else{
+		spin(1.4);//Robô gira no sentido anti-horário
+	}
+}
+
+void Attacker::crossing(Point ball, bool clockwise){
+
+	if(ball.x > get_pose().position.x) {
+		Vector robot_to_ball = ball - get_pose().position;
+		go_to_pose(ball, robot_to_ball);
+
+		if (ball.y > get_pose().position.y && distance(ball, get_pose().position) < size) {
+			if(clockwise)
+			   spin(-1.4);
+            else
+				spin(1.4);
+		}else {
+			go_to_pose(ball, robot_to_ball);
+		}
+	}else{
+
+		uvf_to_goal(ball);
+		//Se ele tiver a frente da bola, faz apenas o UVF,
+		//porém a condição tem que ser verificada pela estratégia antes de chamar o crossing,
+		//ele vai só entrar nesse else em caso de ser chamado em hora indevida.
+	}
+}
+
+void Attacker::atk_mindcontrol(Vector ball_to_goal){
```

Comment:
Se essa for a função de torar o pau, seria mais interessante que ela recebesse apenas a posição da bola e dentro dela se calculasse este Vector ball_to_go para deixar o código da estratégia mais como estratégia e menos como matemática. A posição do gol poderia ser criado no Field.h caso ainda não exista.